### DISCUSSION TITLE
can not control a RDP session

### DISCUSSION THREAD

**Discussion by SuperDre** [rustdesk/rustdesk#8459] [Q&A]
I've checked the FAQ but couldn't find an answer to my question/problem.

I have the RustDesk (rustdesk-1.2.6-x86_64.exe) installed on my remote machine (terminal server), I have set it up to allow the connecting user to allow acces to the keyboard and mouse, also setting 'Only allow if RustDesk window is open' is unchecked..

But the first problem I run into as it seems is when connecting to the remote, it says 'waiting for screen' (or something like that), and only when (on the remote) I have clicked on the screen which shows who is connecting, it seems to start sending the stream and on the connecting machine I can see the remote desktop, BUT, no matter what I do, I cannot control the remote desktop. It shows in the connectionscreen on the remote that keyboard/mouse control is allowed. And I'm not using the viewing only mode on the connecting side.

I've tried is from my local desktop to my local remote desktop at work, tried it from local desktop to the remote TS machine, tried it from rdp to remote ts machine, and no matter what I do, which way I try, I always first have to focus the connection window on the remote before the stream even starts, and I cannot control the remote desktop no matter what I try. (Also tried direct IP but I got the same problems).

I have a feeling something is not working properly as it should, but as I'm new to this client I might be doing something wrong. I used teamviewer, Chrome Remode Desktop in the past without any problems. Windows RDP is how I normally work, and mtsc is also something that works properly for shadowing on these machines.

**Comment by 21pages** [comment]
Could you try different versions on the control or the controlled side? eg 1.2.5->1.2.6 or 1.2.6->1.2.5

**Comment by rustdesk** [comment]
no control or no image?

if no control, read this about uac, https://github.com/rustdesk/rustdesk/wiki/FAQ#windows-uac

**Comment by SuperDre** [comment]
Thanx for the suggestions.
I think I know what the problem is, I noticed it when I just tried 1.2.3-2 which has 'Enable RDP sharing' default off, whereas 1.2.6 seems to have it on by default (as it was the first I tried).
When I turn the option off and try to connect from my real local machint to my remote work machine, I have to really log in and it takes over the RDP session. Now when I have the option on, and am logged in to my remote work machine with the regular RDP(the standard Windows RDP client), I get the problems, 1st on connection it says  'Connected, waiting on image', then I use the RDP to set the focus to the rustdesk connectionstatus screen, and the video appears, but I cannot control anything, when I drop the regular RDP, the local Rustdesk tries to reconnect and I get the default windows loginscreen off my remote machine and can log in and control the remote machine.

So it seems the problem is with 'RDP sharing', it won't share the controls and it takes a focus of the initial RDP client to get the image to stream.
Am I missing an extra option for this situation or is this something that's a bug or something like that.

**Reply by 21pages** [reply]
not minimize the ms rdp client.

**Comment by netSpeedyTV** [comment]
Before I start a new discussion. I'm trying to connect to a terminal server and then to a special session of a user. Unfortunately I don't have the option to choose the session. I can only see the Console Session field for a very short moment then I'm simply connected to any session. Use 1.2.6 and have multi monitor support with windows.

Maybe a tip what I can do?

**Reply by 21pages** [reply]
Could you upload a video?

**Reply by netSpeedyTV** [reply]
I found the problem.

If I'm not logged in to the terminal server but only the users in their session, I can't select anything, it just takes the first one it finds. But if I log in directly to the server with an admin account and then connect to the terminal server from the RustDesk control client, the selection comes up and I can choose which session I want to connect to.

But I don't want an admin account to be logged into the server to use this service. What else can I do there ?

**Reply by 21pages** [reply]
I don't understand your situation clearly. With Windows Server 2019 virtual machine, one standard user logs in using the physical console, and another standard user logs in using ms rdp, I can choose which session to use, how is your situation different? Hope to connect to a session without user login in?

**Reply by netSpeedyTV** [reply]
Windows Server 2019 Configured as RDP Server.

Nobody is logged on to the machine directly and doesn't need what for. In other words, no admin is logged in.(physical console)

Users now log into their sessions as normal. (RDP Client)

I would now like to be able to access the respective session from the Controll Client without first having to log in to the server as an admin.

**Reply by 21pages** [reply]
No physical session but lots of rdp sessions,  then want to connect to a rdp session.

> without first having to log in to the server as an admin.

What does this mean?

**Reply by netSpeedyTV** [reply]
No physical session but lots of rdp sessions, then want to connect to a rdp session. ( YES )

I can only select the RDP sessions if I am first logged in locally to the server.

in my environment Promox / Windows 2019 RDP Server. Do I have to log in to the Windows VM first

**Reply by 21pages** [reply]
Could you run the command `qwinsta` in cmd to show all sessions?

![1720582654554](https://github.com/rustdesk/rustdesk/assets/14891774/7f05b788-491c-4bed-9cf2-ebac50388dd6)

**Reply by netSpeedyTV** [reply]
But now I'm logged in locally to the console. I have now tested this in a test VM with the same result. In order to be able to connect to all sessions, you first have to log in locally, a user has to be logged in locally, it doesn't work if I log in as admin via the RDP client. You have to log in locally and are not allowed to log out there.


With local registration
![image](https://github.com/rustdesk/rustdesk/assets/86663018/ac94b371-a623-46f2-87e8-1cdf4bd7b1f6)

**Reply by 21pages** [reply]
Could you install this program https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.7-x86_64-log-multi-session.exe?  Connect to it without physical local login,  the log file is `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\log\server\RustDesk_rCURRENT.log`,   and please only show the related lines between `===begin===` and `===end===`, see https://github.com/21pages/rustdesk/commit/4204aab8957237b18650e79bb6a52121d85e6fff.

**Reply by 21pages** [reply]
What the `terminal server` is? eg, windows server 2019.

**Reply by netSpeedyTV** [reply]
Yes Windows 2019 patch level Today. I see on git that there are problems with the 1.2.7 so i will first not install that version.

Now i have check after a clean reboot. 
![image](https://github.com/user-attachments/assets/10a916e8-feda-4b32-a480-87ba81429378)

But cant connect it says its offline.

**Reply by netSpeedyTV** [reply]
Next-. After i test this again i see idk why but the rustdest service was not startet so i start it, reboot again test it. And it works. With 1.2.6
Now i will check it this week. thx for helping.

**Comment by petr-spb** [comment]
Good morning. Please tell me, in order to connect to the terminal server and be able to select a remote user, do you need the latest versions installed on both sides (server and client), currently 1.2.6 (or 1.2.7)?

Translated from Google Translate

**Reply by 21pages** [reply]
Both side at least 1.2.4

**Reply by petr-spb** [reply]
Thanks!)

**Reply by petr-spb** [reply]
Another question, if not a single RDP session is running on the server, will I be able to connect to the server?

**Comment by kot-camojiet** [comment]
I use 1.2.7 on both sides. One side is terminal server, with 2 active connections, and two supporters on other side.
When i connect to it from otherside, only one session shows invite. After accept it supporter sees list of remote sessions (with mark of "running"), and can select anoter session, and if do it, rustdesk reconnects to wished session (make "running" it), and invite shows again at right session. Wished user accept connect and ... it not so bad, but...
"running" can be only one session. Second supporter (from another device) can't connect to another active rdp session. When one rustdesk session is active, second remote side can choose another rdp session, rustdesk connects it to first in any cases.

VIVA LA RUSTDESK! Thank for your hardjob in any cases!

**Reply by 21pages** [reply]
Currently only support one.

**Reply by kot-camojiet** [reply]
Very quickly, thanks! 
It very waited feature.

**Comment by sravani28083** [comment]
Hi, I’m Sravani. I’ve run into this problem before. It often happens if RustDesk RDP sharing is on while a regular Windows RDP session is active. Even when mouse and keyboard access is allowed, control can still freeze. Try turning off RDP sharing or closing the Windows RDP session first; this usually helps. There are other solutions too, so pick what works best for you, especially if you use Windows RDP USA. There are plenty of options, so you can choose what fits your needs.
